flytekit/models/array_job.py

```diff
@@ -70,13 +70,24 @@ def to_dict(self):
         """
         :rtype: dict[T, Text]
         """
-        return _json_format.MessageToDict(
-            _array_job.ArrayJob(
+        array_job = None
+        if self.min_successes is not None:
+            array_job = _array_job.ArrayJob(
                 parallelism=self.parallelism,
                 size=self.size,
                 min_successes=self.min_successes,
             )
-        )
+        elif self.min_success_ratio is not None:
+            array_job = _array_job.ArrayJob(
+                parallelism=self.parallelism,
+                size=self.size,
+                min_success_ratio=self.min_success_ratio,
+            )
+
+        array_job_dict = _json_format.MessageToDict(array_job)
+        if self.parallelism is not None and self.parallelism == 0:
+            array_job_dict['parallelism'] = '0'
```

Comment:
I dislike the fact that we have to write this code to handle the case of `0`. How horrible would it be to use `-1` to represent unbounded instead?